src/govuk/components/details/details.js

```diff
@@ -11,58 +11,24 @@ import { generateUniqueID } from '../../common.js'
 var KEY_ENTER = 13
 var KEY_SPACE = 32
 
-// Create a flag to know if the browser supports navtive details
-var NATIVE_DETAILS = typeof document.createElement('details').open === 'boolean'
-
 function Details ($module) {
   this.$module = $module
 }
 
-/**
-* Handle cross-modal click events
-* @param {object} node element
-* @param {function} callback function
-*/
-Details.prototype.handleInputs = function (node, callback) {
-  node.addEventListener('keypress', function (event) {
-    var target = event.target
-    // When the key gets pressed - check if it is enter or space
-    if (event.keyCode === KEY_ENTER || event.keyCode === KEY_SPACE) {
-      if (target.nodeName.toLowerCase() === 'summary') {
-        // Prevent space from scrolling the page
-        // and enter from submitting a form
-        event.preventDefault()
-        // Click to let the click event do all the necessary action
-        if (target.click) {
-          target.click()
-        } else {
-          // except Safari 5.1 and under don't support .click() here
-          callback(event)
-        }
-      }
-    }
-  })
+Details.prototype.init = function () {
+  // If there is native details support, we want to avoid running code to polyfill native behaviour.
+  var hasNativeDetails = typeof this.$module.open === 'boolean'
 
-  // Prevent keyup to prevent clicking twice in Firefox when using space key
-  node.addEventListener('keyup', function (event) {
-    var target = event.target
-    if (event.keyCode === KEY_SPACE) {
-      if (target.nodeName.toLowerCase() === 'summary') {
-        event.preventDefault()
-      }
-    }
-  })
+  if (!this.$module || hasNativeDetails) {
```

Comment:
if `this.$module` is falsey, would we have already errored on line 20 when we tried to get the `this.$module.open` property? Wondering if we need to guard that right at the top of the function instead.

Reply:
Yeah I think this should be done too, good shout